src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayer.java

```diff
@@ -125,4 +126,14 @@ public boolean onIsPlayerSleeping(EntityPlayer self) {
         }
         return false;
     }
+
+    @Override
+    public boolean affectsSpawning() {
+        return this.affectsSpawning;
```

Comment:
This probably also needs to check whether the player's capabilities include that of spectator mode (tile entities don't tick last I recall).